spec/models/product_importer_spec.rb

```diff
@@ -175,6 +175,27 @@
     end
   end
 
+  describe "when shipping category is missing" do
+    before do
+      csv_data = CSV.generate do |csv|
+        csv << ["name", "producer", "category", "on_hand", "price", "units", "unit_type", "variant_unit_name", "on_demand", "shipping_category"]
+        csv << ["Shipping Test", "User Enterprise", "Vegetables", "5", "3.20", "500", "g", "", nil, nil]
+      end
+      File.write('/tmp/test-m.csv', csv_data)
+      file = File.new('/tmp/test-m.csv')
+      settings = {'import_into' => 'product_list'}
+      @importer = ProductImport::ProductImporter.new(file, admin, start: 1, end: 100, settings: settings)
```

Comment:
I not fan of moving the subject of the test, `@importer`, far away from the actual test and I would also make use of RSpec features for consistency. So it can easily be DRYed up by as follows:

```
let(:file) { File.new('/tmp/test-m.csv') }
let(:settings) { {'import_into' => 'product_list'} }
let(:importer) { ProductImport::ProductImporter.new(file, admin, start: 1, end: 100, settings: settings) }

around(:each) do |example|
  File.write('/tmp/test-m.csv', csv_data)
  example.run
  File.delete('/tmp/test-m.csv')
end

describe do
  let(:csv_data) do
    CSV.generate {}
  end

  it 'whatever' do
  end
end
```